components/automate-cli/pkg/verifyserver/models/batchcheck.go

```diff
@@ -324,23 +322,20 @@ func (c *Config) populateAwsCerts(haConfig *config.HaDeployConfig) {
 
 func (c *Config) populateAwsManagedServicesConfig(haConfig *config.HaDeployConfig) {
 	awsManagedServicesConfig := haConfig.Aws.Config
-	c.ExternalPG.PGDbUserName = awsManagedServicesConfig.ManagedRdsDbuserUsername
-	c.ExternalPG.PGDbUserPassword = awsManagedServicesConfig.ManagedRdsDbuserPassword
-	c.ExternalPG.PGInstanceURL = awsManagedServicesConfig.ManagedRdsInstanceURL
-
-	// pg root-ca might be nil in pre deploy state
-	c.ExternalPG.PGRootCert = awsManagedServicesConfig.ManagedRdsCertificate
-	c.ExternalPG.PGSuperuserName = awsManagedServicesConfig.ManagedRdsSuperuserUsername
-	c.ExternalPG.PGSuperuserPassword = awsManagedServicesConfig.ManagedRdsSuperuserPassword
+	c.ExternalPG = &ExternalPG{
+		PGDbUserName:        awsManagedServicesConfig.ManagedRdsDbuserUsername,
+		PGDbUserPassword:    awsManagedServicesConfig.ManagedRdsDbuserPassword,
+		PGInstanceURL:       awsManagedServicesConfig.ManagedRdsInstanceURL,
+		PGRootCert:          awsManagedServicesConfig.ManagedRdsCertificate,
+		PGSuperuserName:     awsManagedServicesConfig.ManagedRdsSuperuserUsername,
+		PGSuperuserPassword: awsManagedServicesConfig.ManagedRdsSuperuserPassword,
+	}
 
 	c.ExternalOS.OSDomainName = awsManagedServicesConfig.ManagedOpensearchDomainName
```

Comment:
Why this should not be changed as above ? to assign c.ExternalOS with new Instance of ExternalOS

Reply:
Done with the new approach